engine/utils/cortex_utils.h

```diff
@@ -31,206 +24,9 @@ inline std::string logs_folder = "./logs";
-        files.push_back(ent->d_name);
-      }
-    }
-    closedir(dir);
-  }
-#endif
-
-  return files;
-}
-
-inline std::string generate_random_string(std::size_t length) {
-  const std::string characters =
-      "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
-
-  std::random_device rd;
-  std::mt19937 generator(rd());
-
-  std::uniform_int_distribution<> distribution(
-      0, static_cast<int>(characters.size()) - 1);
-
-  std::string random_string(length, '\0');
-  std::generate_n(random_string.begin(), length,
-                  [&]() { return characters[distribution(generator)]; });
-
-  return random_string;
-}
-
-#if (defined(__GNUC__) || defined(__clang__)) && \
-    (defined(__x86_64__) || defined(__i386__))
-#include <cpuid.h>
-inline bool isAVX2Supported() {
-  unsigned eax, ebx, ecx, edx;
-  if (__get_cpuid_max(0, nullptr) < 7)
-    return false;
-
-  __get_cpuid_count(7, 0, &eax, &ebx, &ecx, &edx);
-  return (ebx & (1 << 5)) != 0;
-}
-#elif defined(_MSC_VER) && defined(_M_X64) || defined(_M_IX86)
-#include <intrin.h>
-inline bool isAVX2Supported() {
-  int cpuInfo[4];
-  __cpuid(cpuInfo, 0);
-  int nIds = cpuInfo[0];
-  if (nIds >= 7) {
-    __cpuidex(cpuInfo, 7, 0);
-    return (cpuInfo[1] & (1 << 5)) != 0;
-  }
-  return false;
-}
-#else
-inline bool isAVX2Supported() {
-  return false;
-}
-#endif
-
-inline void nitro_logo() {
-  std::string rainbowColors[] = {
-      "\033[93m",  // Yellow
-      "\033[94m",  // Blue
-  };
-
-  std::string resetColor = "\033[0m";
-  std::string asciiArt =
-      "      ___                                   ___           ___     \n"
-      "     /__/        ___           ___        /  /\\         /  /\\    \n"
-      "     \\  \\:\\      /  /\\         /  /\\      /  /::\\       /  /::\\  "
-      " \n"
-      "      \\  \\:\\    /  /:/        /  /:/     /  /:/\\:\\     /  /:/\\:\\ "
-      " \n"
-      "  _____\\__\\:\\  /__/::\\       /  /:/     /  /:/  \\:\\   /  /:/  "
-      "\\:\\ \n"
-      " /__/::::::::\\ \\__\\/\\:\\__   /  /::\\    /__/:/ /:/___ /__/:/ "
-      "\\__\\:\\\n"
-      " \\  \\:\\~~\\~~\\/    \\  \\:\\/\\ /__/:/\\:\\   \\  \\:\\/:::::/ \\  "
-      "\\:\\ /  /:/\n"
-      "  \\  \\:\\  ~~~      \\__\\::/ \\__\\/  \\:\\   \\  \\::/~~~~   \\  "
-      "\\:\\  /:/ \n"
-      "   \\  \\:\\          /__/:/       \\  \\:\\   \\  \\:\\        \\  "
-      "\\:\\/:/  \n"
-      "    \\  \\:\\         \\__\\/         \\__\\/    \\  \\:\\        \\  "
-      "\\::/   \n"
-      "     \\__\\/                                 \\__\\/         \\__\\/    "
-      "\n";
-
-  int colorIndex = 0;
-
-  for (char c : asciiArt) {
-    if (c == '\n') {
-      std::cout << resetColor << c;
-      colorIndex = 0;
-    } else {
-      std::cout << rainbowColors[colorIndex % 2] << c;
-      colorIndex++;
-    }
-  }
-
-  std::cout << resetColor;  // Reset color at the endreturn;
+inline std::string rtrim(const std::string& str) {
```

Comment:
maybe we can remove this function or move it to `string_utils`

Reply:
updated